jenkins-scripts/lib/check_graphic_card.bash

```diff
@@ -3,16 +3,19 @@ GRAPHIC_CARD_PKG=""
 
 export_display_variable()
 {
-    # Hack to found the current display (if available) two steps:
-    # Check for /tmp/.X11-unix/ socket and check if the process is running
-    for i in $(ls /tmp/.X11-unix/ | sed -e 's@^X@:@')
+    # Check for an active X11 display socket.
+    for i in $(find /tmp/.X11-unix -type s)
     do
-      # grep can fail so let's disable the fail or error during its call
-      set +e
-      ps aux | grep bin/X.*$i | grep -v grep
-      set -e
-      if [ $? -eq 0 ] ; then
-	export DISPLAY=$i
+      # If a process is running with the open socket then
+      # lsof will exit successfully.
+      if lsof -Fp $i ; then
+        # Strip the path and leading X from the X11 socket
+        # but check that the resulting string is numeric and
+        # non-empty before exporting.
+        DISPLAY=:$(basename $i | sed -n -E 's/^X([0-9]+)/:\1/p')
```

Comment:
`$i` is no longer just the value of the `DISPLAY` variable so once confirming a likely active X11 server we need to get the value of `DISPLAY` for export. Compared with the prior substitution this one suppresses values that do not match `X[0-9]+`.